README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
```

Comment:
> GET an author address for a given name

Meaning generate a new one, right?  Not look up an existing one?  (4-letter shortnames are not unique enough for that)

Reply:
I was imagining that the first time you `GET` the name it'd generate the author, but all subsequent calls would use the existing key so it'd be consistent. Any thoughts on rewording this to be easier to understand? 😅

Reply:
I think there are 2 actions that might deserve separate URLs:

* Generate identity, given 4-letter shortname as input
* Look up existing identity, given the whole long author address (not just the shortname)

Reply:
Hmm, and a way to list all of the browser's stored author addresses?

If we let apps get the secrets:

`GET earthstar://authors/`   // list the saved authors
`GET earthstar://authors/@xmpl.ao4w9fjawo`   // return full keypair with secret
`GENERATE earthstar://authors/@xmpl`    // generate keypair and return it, including secret

If we don't let apps get the secrets:

`GET earthstar://authors/`   // list the saved authors
`GET earthstar://authors/@xmpl.ao4w9fjawo`   // there's nothing interesting for this to return, maybe we don't need it
`GENERATE earthstar://authors/@xmpl`    // generate keypair and return just the address, but save the secret

These URLs don't have a workspace because they're just about browser identity management.  You might also want to list all users in a workspace but I'm not sure what that URL would be...

Earthstar paths are not allowed to start with `/@`, so we can have special-cased paths like this to talk about specific authors within a workspace, maybe to view their profile or something:
`earthstar://+gardening.xxx/@cinn.xxx`

And maybe something like...
`earthstar://+gardening.xxx/@`
to list all authors in the workspace?  🤷

Reply:
Could we both do a `GET @xmpl` and `GET @xmpl.qasdjasdaj`? Personally, I feel like the fewer verbs and paths we introduce the better. 😅

I really like the idea of `/@` to get the list of authors! Maybe we could do `earthstar://@` to get the list of all authors and `earthstar://+` to get a list of all workspaces?

One reason I'm leaning towards `/@` instead of `/authors/` is to stay closer to the `scheme://origin/path` format of URLs. If we have `/authors/` it feels like it should be an origin somehow which is a bit weird to me. 😅 Also opens the door for having more `methods` in the URL which I think would be nice to avoid.

Reply:
These are kind of obscure and hard to remember, you'd have to look these up if you wanted to use them, but it's ok with me.  Reminds me of Perl a bit :)

### Listing
`GET earthstar://+` to list all workspaces, makes sense
`GET earthstar://@' to list... all users in all workspaces? or all of my saved identities?
`GET earthstar://+workspace.xxx/@` to list all users in that workspace (everyone, or just mine?)

### Identity management
Unsure if we're letting apps get at the secrets or keeping them on the browser side.  Might be easier to start with giving it to apps, and build browser memory later.

`GET @xmpl` to generate a new user and save it to the browser, and also return it (with secret?)
`GET @xmpl.xxxxxx`  to return the secret for a saved user?

Reply:
I was thinking `earthstar://@` would be best for identities that have been generated, all encoutered identities could be cool too, but I'm not sure if that'd require us to query all the workspaces (loaded in this session? ever?).

I like the GET stuff for identity. Should it return as JSON?

Reply:
If the GET identity stuff is returning 2 values (address and secret) then JSON seems natural, that's for format used internally throughout Earthstar.

If it only returns one value it's debatable if it would be better as JSON or just a raw string